transports/p2p/connmgr/connmanager.go

```diff
@@ -209,17 +253,25 @@ func (cm *ConnManager) handleFailedConn(c *ConnReq) {
 			cm.Connect(c)
 		})
 	} else if cm.cfg.GetNewAddress != nil {
-		cm.failedAttempts++
-		if cm.failedAttempts >= maxFailedAttempts {
-			cm.log.Debug().Msgf("Max failed connection attempts reached: [%d] "+
-				"-- retrying connection in: %v", maxFailedAttempts,
-				cm.cfg.RetryDuration)
-			time.AfterFunc(cm.cfg.RetryDuration, func() {
-				cm.NewConnReq()
-			})
+		if c.Addr != nil && c.Addr.String() != "" && cm.cfg.BanAddress != nil {
```

Comment:
When actually c.Addr can be nil or ""?

Reply:
In some occasions we get a connection request without an address. For example when the service starts and the peers are not seeded yet, but also sometimes when the app is running for some time. When this happens, it does a nil panic, that's why I implemented this check.

Resolving this is more in a scope of refactor than this task IMO